### PR TITLE
ES 2.x support

### PR DESCRIPTION
**Pull request by royaldark**
```
This commit fixes the breaking changes introduced in ES 2.0. This is
mostly limited to the removal of several features:
 - More Like This API
 - Mapping deletion
 - Index status API
 - Delete by query
 - Facets
```

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -7,7 +7,7 @@
                  [clj-http              "2.0.0" :exclusions [org.clojure/clojure]]
                  [clojurewerkz/support  "1.1.0" :exclusions [com.google.guava/guava]]
                  ;; used by the native client
-                 [org.elasticsearch/elasticsearch "1.7.3"]]
+                 [org.elasticsearch/elasticsearch "2.0.0"]]
```
Comment (royaldark) [inline project.clj]: There are additional breaking changes in 2.2. This commit only gets 2.0 support working.

File: src/clojurewerkz/elastisch/native/conversion.clj [hunk 2]
```diff
@@ -96,20 +82,20 @@
            org.elasticsearch.action.admin.indices.flush.FlushRequest
            org.elasticsearch.action.admin.indices.refresh.RefreshRequest
            org.elasticsearch.action.admin.indices.cache.clear.ClearIndicesCacheRequest
-           org.elasticsearch.action.admin.indices.status.IndicesStatusRequest
            [org.elasticsearch.action.admin.indices.segments IndicesSegmentsRequest IndicesSegmentResponse IndexSegments]
            org.elasticsearch.action.admin.indices.alias.IndicesAliasesRequest
            org.elasticsearch.action.admin.indices.template.put.PutIndexTemplateRequest
            org.elasticsearch.action.admin.indices.template.delete.DeleteIndexTemplateRequest
-           org.elasticsearch.common.hppc.cursors.ObjectObjectCursor
+           com.carrotsearch.hppc.cursors.ObjectObjectCursor
```
Comment (royaldark) [inline src/clojurewerkz/elastisch/native/conversion.clj]: This is a weird one. I think `ObjectObjectCursor` is only imported to remove reflection warnings anyway.

**Comment by royaldark** [comment]
One failing test for me - but it seems to fail on master too.

```
lein test clojurewerkz.elastisch.native-api.aggregations.avg-aggregation-test
Exception in thread "main" java.net.ConnectException: Connection refused, compiling:(/tmp/form-init8299152205120309030.clj:1:73)
	at clojure.lang.Compiler.load(Compiler.java:7239)
	at clojure.lang.Compiler.loadFile(Compiler.java:7165)
	at clojure.main$load_script.invoke(main.clj:275)
	at clojure.main$init_opt.invoke(main.clj:280)
	at clojure.main$initialize.invoke(main.clj:308)
	at clojure.main$null_opt.invoke(main.clj:343)
	at clojure.main$main.doInvoke(main.clj:421)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at clojure.lang.Var.invoke(Var.java:383)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.Var.applyTo(Var.java:700)
	at clojure.main.main(main.java:37)
Caused by: java.net.ConnectException: Connection refused
	at java.net.PlainSocketImpl.socketConnect(Native Method)
	at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:345)
	at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:206)
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:188)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at org.apache.http.conn.scheme.PlainSocketFactory.connectSocket(PlainSocketFactory.java:127)
	at org.apache.http.impl.conn.DefaultClientConnectionOperator.openConnection(DefaultClientConnectionOperator.java:180)
	at org.apache.http.impl.conn.ManagedClientConnectionImpl.open(ManagedClientConnectionImpl.java:294)
	at org.apache.http.impl.client.DefaultRequestDirector.tryConnect(DefaultRequestDirector.java:643)
	at org.apache.http.impl.client.DefaultRequestDirector.execute(DefaultRequestDirector.java:479)
	at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:906)
	at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:805)
	at org.apache.http.impl.client.AbstractHttpClient.execute(AbstractHttpClient.java:784)
	at clj_http.core$request.invoke(core.clj:277)
	at clojure.lang.Var.invoke(Var.java:379)
	at clj_http.client$wrap_request_timing$fn__3021.invoke(client.clj:633)
	at clj_http.client$wrap_lower_case_headers$fn__3018.invoke(client.clj:624)
	at clj_http.client$wrap_query_params$fn__2959.invoke(client.clj:502)
	at clj_http.client$wrap_basic_auth$fn__2964.invoke(client.clj:518)
	at clj_http.client$wrap_oauth$fn__2968.invoke(client.clj:528)
	at clj_http.client$wrap_user_info$fn__2973.invoke(client.clj:541)
	at clj_http.client$wrap_url$fn__3005.invoke(client.clj:599)
	at clj_http.client$wrap_redirects$fn__2812.invoke(client.clj:189)
	at clj_http.client$wrap_decompression$fn__2840.invoke(client.clj:258)
	at clj_http.client$wrap_input_coercion$fn__2908.invoke(client.clj:393)
	at clj_http.client$wrap_additional_header_parsing$fn__2929.invoke(client.clj:436)
	at clj_http.client$wrap_output_coercion$fn__2899.invoke(client.clj:347)
	at clj_http.client$wrap_exceptions$fn__2793.invoke(client.clj:141)
	at clj_http.client$wrap_accept$fn__2939.invoke(client.clj:461)
	at clj_http.client$wrap_accept_encoding$fn__2945.invoke(client.clj:478)
	at clj_http.client$wrap_content_type$fn__2934.invoke(client.clj:454)
	at clj_http.client$wrap_form_params$fn__2983.invoke(client.clj:565)
	at clj_http.client$wrap_nested_params$fn__3001.invoke(client.clj:589)
	at clj_http.client$wrap_method$fn__2978.invoke(client.clj:548)
	at clj_http.cookies$wrap_cookies$fn__1737.invoke(cookies.clj:121)
	at clj_http.links$wrap_links$fn__1774.invoke(links.clj:50)
	at clj_http.client$wrap_unknown_host$fn__3010.invoke(client.clj:608)
	at clj_http.client$delete.doInvoke(client.clj:739)
	at clojure.lang.RestFn.invoke(RestFn.java:423)
	at clojurewerkz.elastisch.rest$delete.invoke(rest.clj:93)
	at clojurewerkz.elastisch.rest.index$delete.invoke(index.clj:80)
	at clojurewerkz.elastisch.fixtures$reset_indexes_STAR_.invoke(fixtures.clj:26)
	at clojurewerkz.elastisch.fixtures$reset_indexes.invoke(fixtures.clj:31)
	at clojure.test$compose_fixtures$fn__7664$fn__7665.invoke(test.clj:681)
	at clojure.test$default_fixture.invoke(test.clj:674)
	at clojure.test$compose_fixtures$fn__7664.invoke(test.clj:681)
	at clojure.test$compose_fixtures$fn__7664.invoke(test.clj:681)
	at clojure.test$test_vars$fn__7692.invoke(test.clj:722)
	at clojure.test$default_fixture.invoke(test.clj:674)
	at clojure.test$test_vars.invoke(test.clj:718)
	at clojure.test$test_all_vars.invoke(test.clj:728)
	at clojure.test$test_ns.invoke(test.clj:747)
	at clojure.core$map$fn__4553.invoke(core.clj:2624)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.Cons.next(Cons.java:39)
	at clojure.lang.RT.next(RT.java:674)
	at clojure.core$next__4112.invoke(core.clj:64)
	at clojure.core$reduce1.invoke(core.clj:909)
	at clojure.core$reduce1.invoke(core.clj:900)
	at clojure.core$merge_with.doInvoke(core.clj:2936)
	at clojure.lang.RestFn.applyTo(RestFn.java:139)
	at clojure.core$apply.invoke(core.clj:632)
	at clojure.test$run_tests.doInvoke(test.clj:762)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at clojure.core$apply.invoke(core.clj:630)
	at user$eval87$fn__151$fn__185.invoke(form-init8299152205120309030.clj:1)
	at user$eval87$fn__151$fn__152.invoke(form-init8299152205120309030.clj:1)
	at user$eval87$fn__151.invoke(form-init8299152205120309030.clj:1)
	at user$eval87.invoke(form-init8299152205120309030.clj:1)
	at clojure.lang.Compiler.eval(Compiler.java:6782)
	at clojure.lang.Compiler.eval(Compiler.java:6772)
	at clojure.lang.Compiler.load(Compiler.java:7227)
	... 11 more
```

File: src/clojurewerkz/elastisch/native/conversion.clj (hunk 2 above)
Comment (michaelklishin) [inline src/clojurewerkz/elastisch/native/conversion.clj]: Sounds about right.

**Comment by michaelklishin** [comment]
@royaldark thank you, merged. There are many more failures, mostly in the native client, and even more so when I upgrade ES Java client to 2.2.x. So we are quite far from making the existing test suite pass but now probably 2/3rds of the way there.

**Pull request merged**